src/api/v1beta1/properties.go

```diff
@@ -130,8 +130,22 @@ func (dk *DynaKube) StatsdImage() string {
 	return resolveImagePath(newStatsdImagePath(dk))
 }
 
+func (dk *DynaKube) NeedsReadOnlyOneAgents() bool {
+	inSupportedMode := dk.HostMonitoringMode() || dk.CloudNativeFullstackMode()
+	return inSupportedMode && dk.FeatureReadOnlyOneAgent()
+}
+
 func (dk *DynaKube) NeedsCSIDriver() bool {
-	return dk.CloudNativeFullstackMode() || (dk.ApplicationMonitoringMode() && dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver != nil && *dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver)
+	if dk.CloudNativeFullstackMode() {
```

Comment:
chaining the checks can look nice if it's formatted correctly, this is pretty confusing/unreadable to me with multiple checks for the same thing